A solution for the first homework (hwdetect)

Renalime opened the pull request:
A bash script that detects whether sd cards, flash drives, i2c devices and usb-ttl convertors are connected. 
Each hardware type is served by a separate function with a descriptive name.
All the caveats are described in commit messages.

yekovalyov committed:
homework: task for hardware detector

Signed-off-by: Yevgen Kovalyov <yevgen.kovalyov@globallogic.com>

Renalime committed:
hwdetect: Add an empty bash script file

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
hwdetect: Add a function to check USB to TTL

Added a function check_usb_to_ttl() to check whether a USB to TTL converter
is attached.

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
hwdetect: Add a function to check flash drives

Added a function check_flash_drives to check whether a USB flash drive
is connected.

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
hwdetect: Add a function to check SD card

Added a function check_sd_card() to check whether an SD card is
connected.
This function assumes that all SD cards are associated with respective
/dev/mmcblk* files, although sometimes cards are mapped to /dev/sd*
files. A solution for that mapping has not been found, yet.

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
hwdetect: Add a function to check i2c devices

Added a function check_i2c_devices() to check whether an I2C device
is connected.
This function uses the i2cdetect utility from the i2c-tools package.

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

yekovalyov commented on hwdetect/hwdetect.sh:
```diff
@@ -0,0 +1,43 @@
+	usb_to_ttl=$(ls /dev | grep -E "^ttyUSB[0-9]+$")
+	if [ -n "$usb_to_ttl" ];
+	then
+		echo "A USB to TTL convertor is connected."
```
Please add device file name into the message.

yekovalyov commented on hwdetect/hwdetect.sh:
```diff
@@ -0,0 +1,43 @@
+check_sd_card() {
+	if [ -n "$(ls /dev | grep -E "^mmcblk[0-9]+$")" ];
+	then
+		echo "An SD card is connected."
```
Please add device file name into the message.

yekovalyov commented on hwdetect/hwdetect.sh:
```diff
@@ -0,0 +1,43 @@
+		i2c_bus=$(i2cdetect -y ${i} | awk '{if (NR!=1) {print}}' | awk '{$1=""}1')
+		if [ -n "$(echo $i2c_bus | grep '[0-9]')" ];
+		then
+			echo "An i2c device is connected"
```
Please add i2c bus number.

Renalime committed:
hwdetect: Remove unnecessary quotes in loop literal

Removed the quotes around loop literal in check_flash_drives(), because
it makes bash treat the literal as a single string, not as a list.

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
hwdetect: Add a flash drive device file in message

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
hwdetect: Add USB to TTL device name into message

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
hwdetect: Add an SD card device file into message

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime committed:
hwdetect: Add an i2c device file into message

Signed-off-by: Andrii Revva <milaner.work@gmail.com>

Renalime commented:
@yekovalyov Done all the requested features, although it doesn't look pretty, since sd card and usb to ttl functions basically spits out all the devices that were found by grep. But it works.

Renalime commented:
@yekovalyov One more thing, should I remove 'question' label after the requested features are ready?
